### ISSUE TITLE
Cannot install botocore 1.25.12 with pip

### ISSUE THREAD

**Issue by bencehornak-gls** [boto/botocore#2671]
### Describe the bug

When I run `pip install botocore==1.25.12` I get an error:

```
Could not find a version that satisfies the requirement botocore==1.25.12 (from versions: 0.4.1, 0.4.2, ..., 1.25.11)
```

### Expected Behavior

It should be installed without warning.

### Current Behavior

```
pip install botocore==1.25.12 -v
```

[pip.log](https://github.com/boto/botocore/files/8679673/pip.log)

Based on the logs pip collects the available versions from this URL: https://pypi.org/simple/botocore/. When I open this website manually, I also don't see the mentioned version.

Versions on this page:

```
...
botocore-1.25.9-py3-none-any.whl
botocore-1.25.9.tar.gz
botocore-1.25.10-py3-none-any.whl
botocore-1.25.10.tar.gz
botocore-1.25.11-py3-none-any.whl
botocore-1.25.11.tar.gz
botocore-1.25.12-py3-none-any.whl
```

Note that `botocore-1.25.12.tar.gz` is missing.

I don't know in which system the error lies, but you might have some control over it.

### Reproduction Steps

```sh
pip install botocore==1.25.12
```

### SDK version used

1.25.12

### Environment details (OS name and version, etc.)

Centos7
Python 2.7.5
pip 8.1.2

**Comment by nateprewitt** [comment]
Hi @bencehornak-gls, this is a bug in the CDN fronting PyPI. There's an open tracking ticket in Warehouse (pypa/warehouse#10390), the system that PyPI is built on. We unfortunately have no control over fixing this. It will be resolved either when the CDN cache refreshes, or when we publish a new package (which is generally every weekday by 13:00 PT).

What's interesting here is that the wheel is available but `pip` isn't using it. All versions of `pip` released with Python 3.7+ should be defaulting to using the wheel. Could you provide more information about your environment (Python version, pip version, any global pip settings you've changed)?

**Comment by bencehornak-gls** [comment]
Hi @nateprewitt, thanks for the insightful answer. You were right, today it started to work. Very frustrating CDN behavior, which makes our Chef provisioning totally unreliable.

Do you have any tips how I could work around this? In your experience would it help to fix our `awscli` (and thus `botocore`) to a version which is consistently accessible through the CDN?

Regarding my environment, I updated the description with the exact versions. I am using Python 2.7 due to external constraints.

**Comment by nateprewitt** [comment]
Ah, I believe you have a few options. If you're using Python 2.7, we haven't released a version of Boto3 or the CLI that's compatible since July of 2021. Pinning your installs to `awscli<1.20` and `botocore<1.21` (`boto3<1.18` if needed) should stop pip from trying to grab the latest version and backtracking.

A second fix is pip 9.0.2 and later should never try to grab this version of botocore for you. I would recommend trying to upgrade that if possible as it should mitigate the issue as well.

Let us know if you have any other questions or comments on the above options.

**Comment by bencehornak-gls** [comment]
Thanks a lot @nateprewitt, I will try it with your instructions. I think at this point it is safe to close this issue, everything needed is here.
